tests/Validator/ValuesOfCorrectTypeTest.php

```diff
@@ -1649,4 +1650,42 @@ public function testListVariablesWithInvalidItem() : void
 
         self::assertTrue($errors[0]->isClientSafe());
     }
+
+    /**
+     * @see it('error messages can be overwriten')
+     */
+    public function testOverwriting() : void
+    {
+        DocumentValidator::addRule(
+            new class() extends ValuesOfCorrectType
+            {
+                public function getName()
+                {
+                    return (new ValuesOfCorrectType())->getName();
```

Comment:
Why is this function overwritten at all?

Reply:
Because `getName()` used by `DocumentValidator::addRule()` and we want to replace `ValuesOfCorrectType` completely.

Reply:
And the default realization of `getName()` uses `static::class`

https://github.com/webonyx/graphql-php/blob/65428ce4da4ba6918a20f8f638345e894d2575d7/src/Validator/Rules/ValidationRule.php#L16-L19

Reply:
That seems like an annoying gotcha for users that extend this class.

Reply:
🤷‍♂️ Maybe something like?

```
class ValuesOfCorrectType extends ValidationRule
{
    // (1)
    /** @var string */
    protected $name = self::class;
    
    // (2) or this (probably a breaking change) 
    public function __construct() { 
        $this->name = self::class;
    }
```